system/critmon/critmon.c

```diff
@@ -378,101 +378,115 @@ static void critmon_global_crit(void)
+          return EXIT_FAILURE;
         }
+    }
 
-      /* Read each directory entry */
+  /* Read each directory entry */
 
-      for (; ; )
+  for (; ; )
+    {
+      FAR struct dirent *entryp = readdir(dirp);
+      if (entryp == NULL)
         {
-          FAR struct dirent *entryp = readdir(dirp);
-          if (entryp == NULL)
-            {
-              /* Finished with this directory */
+          /* Finished with this directory */
 
-              break;
-            }
+          break;
+        }
 
-          /* Task/thread entries in the /proc directory will all be (1)
-           * directories with (2) all numeric names.
-           */
+      /* Task/thread entries in the /proc directory will all be (1)
+       * directories with (2) all numeric names.
+       */
 
-          if (DIRENT_ISDIRECTORY(entryp->d_type) &&
-              critmon_check_name(entryp->d_name))
+      if (DIRENT_ISDIRECTORY(entryp->d_type) &&
+          critmon_check_name(entryp->d_name))
+        {
+          /* Looks good -- process the directory */
+
+          ret = critmon_process_directory(entryp);
+          if (ret < 0)
             {
-              /* Looks good -- process the directory */
+              /* Failed to process the thread directory */
 
-              ret = critmon_process_directory(entryp);
-              if (ret < 0)
-                {
-                  /* Failed to process the thread directory */
+              fprintf(stderr, "Csection Monitor: "
+                      "Failed to process sub-directory: %s\n",
+                      entryp->d_name);
 
+              if (++errcount > 100)
+                {
                   fprintf(stderr, "Csection Monitor: "
-                          "Failed to process sub-directory: %s\n",
-                          entryp->d_name);
-
-                  if (++errcount > 100)
-                    {
-                      fprintf(stderr, "Csection Monitor: "
-                              "Too many errors ... exiting\n");
-                      exitcode = EXIT_FAILURE;
-                      break;
-                    }
+                          "Too many errors ... exiting\n");
+                  exitcode = EXIT_FAILURE;
+                  break;
                 }
             }
         }
+    }
+
+  closedir(dirp);
+  fputc('\n', stdout);
+  return exitcode;
+}
+
+/****************************************************************************
+ * Name: critmon_daemon
+ ****************************************************************************/
+
+static int critmon_daemon(int argc, char **argv)
+{
+  int exitcode = EXIT_SUCCESS;
+
+  printf("Csection Monitor: Running: %d\n", g_critmon.pid);
+
+  /* Loop until we detect that there is a request to stop. */
+
+  while (!g_critmon.stop)
+    {
+      /* Wait for the next sample interval */
 
-      closedir(dirp);
-      fputc('\n', stdout);
+      sleep(CONFIG_SYSTEM_CRITMONITOR_INTERVAL);
```

Comment:
Should this move to under the exit check so the user gets more immediate feedback?

Reply:
Done.